### PR TITLE
added parameter -e <NAME> <VALUE> for SpoonRunner.

### PR DESCRIPTION
**Pull request by roshakorost**
Hello.

It might be useful to use test runners with "extra" additional parameters.
 
E.g. I need to run spoon only for functional tests (These are tests that derive from InstrumentationTestCase) 
InstrumentationTestRunner allows to do this, using '-e func true' 
here -e paramater set the 'func' argument with value 'true'

However, currently it is not possible to pass 'extra' argument to test runner.

This commit adds -e <NAME> <VALUE> parameter to command line arguments of spoon-runner, and to SpoonRunner.Builder so it will be possible to pass parameters to the runner.

### CODE REVIEW

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java [hunk 1]
```diff
@@ -336,6 +340,13 @@ public Builder addTestRunListener(ITestRunListener testRunListener) {
       return this;
     }
 
+    public Builder setRunnerArg(String name, String value) {
+        checkNotNull(name, "Runner arg name cannot be null");
```
Comment (JakeWharton) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java]: two space indent

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java [hunk 2]
```diff
@@ -403,6 +414,11 @@ public SpoonRunner build() {
 
     @Parameter(names = { "-h", "--help" }, description = "Command help", help = true, hidden = true)
     public boolean help;
+
+    @Parameter(names = {"-e" }, arity = 2,
+            description = "-e <NAME> <VALUE>: Set instrument runner argument <NAME> to <VALUE>")
```
Comment (JakeWharton) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java]: continuations are four space indent

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java [hunk 3]
```diff
@@ -461,10 +477,16 @@ public static void main(String... args) {
...
+      if (parsedArgs.runnerArg != null) {
+        String key = parsedArgs.runnerArg.get(0);
+        String value = parsedArgs.runnerArg.get(1);
+        spoonRunnerBuilder.setRunnerArg(key, value);
+      }
+      SpoonRunner spoonRunner = spoonRunnerBuilder.build();
+
 
-    if (!spoonRunner.run() && parsedArgs.failOnFailure) {
+      if (!spoonRunner.run() && parsedArgs.failOnFailure) {
```
Comment (JakeWharton) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java]: revert

Comment (edenman) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java]: revert this

**Comment by JakeWharton** [comment]
Are you only able to set one key/value pair? I would think you could set multiple of them.

**Comment by roshakorost** [comment]
d It is not difficult to add support of multiple key/value pairs. 

The only one problem is with Jcomander, because I think syntax for setting multiple key/values should be -e key1 value1 -e key2 value2 ... . But I didn't find way to do that using Jcomander.

**Comment by edenman** [comment]
Closing: #266 solves this.

**Pull request closed without merging**